### ISSUE TITLE
EIN completion / help is inactive in ein:edit-cell-contents

### ISSUE THREAD

**Issue by cpbotha** [millejoh/emacs-ipython-notebook#183]
When I invoke ein:edit-cell-contents (`C-c '`) to enter the EinCell minor mode, and then do `M-x ein:completer-complete`, I get the following error:

`((cl-assertion-failed (ein:kernel-live-p kernel) "complete_reply: Kernel is not active."))`

The hot-key `C-c C-i` is not bound in EinCell, when it usually is in the notebook.

Also, `ein:pytools-request-tooltip-or-help` yields the error `(error "No kernel related to the current buffer.")` (and `C-c C-f` is also not bound, when it usually is in the notebook itself)

When I explicitly connect the EinCell buffer to the notebook in question with `ein:connect-to-notebook-command` completion and help work (yay!) but now `C-c C-c` has been rebound to `ein:connect-run-or-eval-buffer` and when I invoke it, Emacs asks me: `Directory /home/cpbotha/*EIN Src *ein: http://127.0.0.1:8888/' does not exist; create?`

Usually, `C-c C-c` is bound to `ein:edit-cell-save-and-execute` and executes the code immediately in the context of its cell.

To summarise: I like to use EinCell mode. How can I do that, with `C-c C-c` correctly executing the cell and completion and help working?

I could try hacking around this by adding `connect-to-notebook-command` to the EinCell mode hook, and then rebinding `C-c C-c` to `edit-cell-save-and-execute`, but surely having completion, help and execution, all in notebook context, should work by default?

**Comment by millejoh** [comment]
@cpbotha, the use case for ein edit cells is to work around the fact that mixing Python and Markdown major modes in the same notebook buffer has proven difficult to implement. Think something along the line of org mode's method of editing source code blocks.

What ein edit cell does is create a new buffer with the appropriate major mode enabled (at the moment this is either Markdown or Python). If autocompletion isn't working then one solution would be to check how your Python major mode is configured in general.

It still might be a good idea to set an appropriate buffer local variable to the running kernel when creating the edit buffer so things like `ein:completer-complete` work correctly, though I'll need to be careful that doing so doesn't interfere with how Python mode does autocompletion. That say, I don't want to inadvertently break someone's setup for using, say, anaconda-mode with python-mode.

**Comment by cpbotha** [comment]
Understood, thanks for the explanation.

Your idea of having a buffer local variable pointing to the running kernel in EinCell mode would be great. This would enable users to be able to opt-in to ein completion and tooltips. Personally I use elpy and company-mode for my Python editing, but in an EinCell buffer this is not as useful, because it's not EIN aware.

**Comment by cpbotha** [comment]
I have made a pull request whereby ein:get-kernel can retrieve the source worksheet kernel in edit-cell-mode. This means that interactive functions such as ein:completer-complete now work in edit-cell-mode, and can easily be rebound by the user if they so prefer.

https://github.com/millejoh/emacs-ipython-notebook/pull/186

**Comment by cpbotha** [comment]
PR 186 has been merged.
